### PR TITLE
S3 static file serving/Django-pipeline & Compressing with Yugilfy

### PR DESCRIPTION
**Pull request by kave**
S3 static file serving integration with Django-pipeline & Compressing with Yugilfy for Prod env's

### CODE REVIEW

File: face_it/settings/base.py [hunk 1]
```diff
@@ -89,12 +81,54 @@
...
+
+# URL that handles the media served from MEDIA_ROOT. Make sure to use a
+# trailing slash.
+# Examples: "http://media.lawrence.com/media/", "http://example.com/media/"
+MEDIA_URL = STATIC_URL + 'media/'
+
+STATICFILES_STORAGE = 'pipeline.storage.PipelineStorage'
+PIPELINE_CSS_COMPRESSOR = 'pipeline.compressors.NoopCompressor'
+PIPELINE_JS_COMPRESSOR = 'pipeline.compressors.NoopCompressor'
+
```
Comment (kave) [inline face_it/settings/base.py:102]: by default not compressing the files

File: face_it/settings/production.py [hunk 2]
```diff
@@ -14,3 +14,14 @@
...
+AWS_STORAGE_BUCKET_NAME = os.environ['AWS_STORAGE_BUCKET_NAME']
+
+STATICFILES_STORAGE = 'core.storage.S3PipelineManifestStorage'
+STATIC_URL = 'http://s3.amazonaws.com/%s/' % AWS_STORAGE_BUCKET_NAME
+
+AWS_QUERYSTRING_AUTH = False
+AWS_S3_FILE_OVERWRITE = True
+
+PIPELINE_JS_COMPRESSOR = 'pipeline.compressors.yuglify.YuglifyCompressor'
+PIPELINE_CSS_COMPRESSOR = 'pipeline.compressors.yuglify.YuglifyCompressor'
```
Comment (kave) [inline face_it/settings/production.py:27]: using Yuglify in the production environment

File: core/templates/base.html [hunk 3]
```diff
@@ -2,6 +2,7 @@
 {% load compress %}
 {% load replace %}
 {% load staticfiles %}
+{% load pipeline %}
```
Comment (kave) [inline core/templates/base.html:6]: put django-pipeline configurations back into the application to compress CSS/JSS into individual files

File: core/storage.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+from pipeline.storage import PipelineMixin
+
+from storages.backends.s3boto import S3BotoStorage
+
+class S3PipelineManifestStorage(PipelineMixin, ManifestFilesMixin, S3BotoStorage):
+    pass
+
+class S3PipelineCachedStorage(PipelineMixin, CachedFilesMixin, S3BotoStorage):
+    pass
```
Comment (kave) [inline core/storage.py:11]: S3 related storage classes for django-pipeline integration

**Comment by m3brown** [comment]
This is great!

**Pull request merged**